components/EmotionCard.js

```diff
@@ -3,18 +3,22 @@ import BookmarkButton from "./BookmarkButton";
 import Link from "next/link";
 import EmotionCardContent from "./EmotionCardContent";
 import formatDate from "./TransformDateTime";
-import { emotionList } from "@/lib/emotionList";
 
 export default function EmotionCard({
   emotion,
   onToggleBookmark,
   isBookmarked,
   isDetailsPage = false,
+  emotionTypes,
 }) {
   const { date, time } = formatDate(emotion.dateTime);
 
   function getEmotionColor(type) {
-    const emotion = emotionList.find((item) => item.emotionType === type);
+    if (!emotionTypes || emotionTypes.length === 0) {
+      return "var(--color-frame)";
+    }
```

Comment:
emotionTypes should always be at least an empty array, therefore you should not need this check, `array.find` will return null if the length of the array is 0. Therefore, this function already returns the correct color if no emotionTypes are available.